### PR TITLE
Adding RTL support for DetailsList drag-drop to reorder columns

### LINKED ISSUE #7230: Column drag-drop not working in RTL
### Environment Information 
 
 
- __Package version(s)__: 
- __Browser and OS versions__: All

### Please provide a reproduction of the bug in a codepen:
 

Repro steps:

1. On the Fabric demo app page, click on the gear icon in the top right.
2. Select "Render in RTL"
3. Try dragging and dropping a column header.
4. The drop hint always comes at the first index, no matter where we drag a column header.
5. Reorder happens according to the drop hint index.

 
#### Actual behavior:
 
Drop hint always comes at the first index, no matter where we drag a column header.

image

 
#### Expected behavior:

Drop hint should come according to the position where the column header is being dragged. The behavior should be the same as it is for LTR.

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? Yes
 
Requested priority: (Blocking, High, Normal, Low) Normal
 
Products/sites affected: (if applicable) SharePoint

### PR DESCRIPTION
**Pull request by arkogupta**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #7230 
- [X] Include a change request file using `$ npm run change`

#### Description of changes

This change adds support for dragging and dropping columns to reorder them, in RTL languages.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx [hunk 1]
```diff
@@ -517,14 +536,13 @@ export class DetailsHeaderBase extends BaseComponent<IDetailsHeaderBaseProps, ID
...
-            eventXRelativePosition >= this._dropHintDetails[middleIndex].startX &&
-            eventXRelativePosition <= this._dropHintDetails[middleIndex].endX
+            this._liesBetween(eventXRelativePosition, this._dropHintDetails[middleIndex].startX, this._dropHintDetails[middleIndex].endX)
           ) {
             indexToUpdate = middleIndex;
             break;
-          } else if (eventXRelativePosition < this._dropHintDetails[middleIndex].originX) {
+          } else if (this._isBefore(eventXRelativePosition, this._dropHintDetails[middleIndex].originX)) {
             endIndex = middleIndex;
-          } else if (eventXRelativePosition > this._dropHintDetails[middleIndex].originX) {
```
Comment (natalieethell) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx]: It looks like the original comparison was >, but the new `_isAfter` method checks >=. Is this intentional?

**Comment by natalieethell** [comment]
Hey @arkogupta I just tested out your changes locally and I'm not seeing the right behavior for the DetailsList Drag & Drop example in RTL.

detailslistdragdroprtlnotworking

It looks like the columns are still either jumping to the far left or the far right.

**Comment by KevinTCoughlin** [comment]
Hmmm... unsure why @natalieethell ran into the above 🤔. In my _local_ testing this seems to work well. Here is a test pass in Edge:

dl_drag_and_drop_rtl

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx (hunk 1 above)
Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx]: @arkogupta would you be able to clarify the above?

**Comment by natalieethell** [comment]
Update: got it working! Just synced up with @KevinTCoughlin offline and it works for me when I click the "Render in RTL" option on the gear icon in the top right corner of the demo page. Looks good.

(Before, I was just changing the `<html dir="ltr">` tag at the top of the page to `<html dir="rtl">`. That didn't seem to re-load everything correctly.)

I'll approve this, but could we resolve the comment above before merging? Thanks!

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx (hunk 1 above)
Comment (arkogupta) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx]: I didn't want to add an extra equality check ( `|| eventXRelativePosition  === this._dropHintDetails[...]...`) in lines 508 and 510, to keep it short. So used it in the utility method and reused it here as well, where we don't actually require the equality check. But doesn't make any difference in the logic as the equality case would be filtered by the first if condition anyway. Hence kept it this way.

Comment (natalieethell) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx]: Okay, I think that's fine

**Pull request merged**